scripts/include/helpers.inc.php

```diff
@@ -121,45 +121,97 @@ function mergeXliffDomToAtom(DOMDocument $domA, DOMDocument $domB): DOMDocument
 
 function mergeXliffDomToWeblate(DOMDocument $domA, DOMDocument $domB): DOMDocument
 {
-  // Elements in A but not in B -> copy to B
-  $sourcesA = $domA->getElementsByTagName('source');
-  $sourcesB = $domB->getElementsByTagName('source');
+  // Elements in A AND B -> check source string and if different -> replace in B
+  $transUnitsA = $domA->getElementsByTagName('trans-unit');
+  $transUnitsB = $domB->getElementsByTagName('trans-unit');
 
-  foreach ($sourcesA as $source) {
-    $stringsA[] = $source->nodeValue;
+  $i = 0;
+  foreach ($transUnitsA as $transUnit) {
+    if ($transUnitsA->item($i)->hasAttribute('id')) {
+      $transUnitsA->item($i)->setIdAttribute('id', true);
+
+      $idListA[] = $transUnitsA->item($i)->getAttribute('id');
+    }
+
+    $i++;
   }
 
-  foreach ($sourcesB as $source) {
-    $stringsB[] = $source->nodeValue;
+  $i = 0;
+  foreach ($transUnitsB as $transUnit) {
+    if ($transUnitsB->item($i)->hasAttribute('id')) {
+      $transUnitsB->item($i)->setIdAttribute('id', true);
+
+      $idListB[] = $transUnitsB->item($i)->getAttribute('id');
+    }
+
+    $i++;
   }
 
-  $copyToB = array_diff($stringsA, $stringsB);
+  // Items in A that are also in B.
```

Comment:
Suggestion: this seems like a good seam to add a `updateExistingTransUnits($transUnitsA, $transUnitsB)` (or similar) method.

Reply:
Omitting for now